src/type-translator.ts

```diff
@@ -164,12 +164,24 @@ export class TypeTranslator {
    * - TypeChecker.typeToString translates Array as T[].
    * - TypeChecker.symbolToString emits types without their namespace,
    *   and doesn't let you pass the flag to control that.
+   * @param useFqn whether to scope the name using its fully qualified name.
    */
-  public symbolToString(sym: ts.Symbol): string {
+  public symbolToString(sym: ts.Symbol, useFqn: boolean): string {
     // This follows getSingleLineStringWriter in the TypeScript compiler.
     let str = '';
     let alias = this.symbolsToAliasedNames.get(sym);
     if (alias) return alias;
+    if (useFqn) {
+      const fqn = this.typeChecker.getFullyQualifiedName(sym);
+      if (!fqn.startsWith(`"`) && !fqn.startsWith(`'`)) {
```

Comment:
Can you comment here about what it means for a fqn to be quoted?  I don't quite follow.